android/app/src/main/java/com/zulipmobile/MainActivity.kt

```diff
@@ -39,7 +39,7 @@ open class MainActivity : ReactActivity() {
         maybeHandleShareIntent(intent, application as ReactApplication, contentResolver)
     }
 
-    override fun onNewIntent(intent: Intent?) {
+    override fun onNewIntent(intent: Intent) {
```

Comment:
android notif [nfc]: Make MainActivity.onNewIntent's `intent` arg required
    
    ReactActivity's `.onNewIntent`, which we're overriding, requires `intent`.

I think it doesn't. Here's the declaration in `ReactActivity.java`:
```java
  @Override
  public void onNewIntent(Intent intent) {
```
That doesn't mention the possibility of the argument being null. But... in Java, all object types are nullable. So Java `Intent` translates to Kotlin `Intent?`.

Sometimes there's a Java annotation `@NotNull` indicating that a given value isn't supposed to be null. But there doesn't seem to be one of those here. So if a caller passes null, no linter would complain either.

If you look in the Kotlin version of the SDK docs:
https://developer.android.com/reference/kotlin/android/app/Activity#onnewintent
this is reflected in the notation `Intent!`:
> `protected open fun onNewIntent(intent: Intent!): Unit`

The meaning of that `!` notation is defined here:
https://kotlinlang.org/docs/java-interop.html#notation-for-platform-types
> `T!` means "`T` or `T?`"

Practically, the way I ended up with this version of the definition is probably that I started typing "onNewIntent" and let Android Studio autocomplete an empty override for me.